system/src/system_task.cpp

```diff
@@ -478,7 +478,7 @@ extern void system_handle_button_clicks(bool isIsr);
 
 void Spark_Idle_Events(bool force_events/*=false*/)
 {
-    HAL_Notify_WDT();
+    hal_watchdog_kick();
```

Comment:
I'd argue that we need to remove all these calls from the system right now and leave the watchdog handling up to the application entirely.